Signal/src/Profiles/OWSProfileManager.m

```diff
@@ -156,32 +150,30 @@ - (instancetype)initWithStorageManager:(TSStorageManager *)storageManager
     OWSAssert([NSThread isMainThread]);
     OWSAssert(storageManager);
     OWSAssert(messageSender);
+    OWSAssert(messageSender);
 
     _messageSender = messageSender;
     _dbConnection = storageManager.newDatabaseConnection;
+    _networkManager = networkManager;
+
     _userProfileWhitelistCache = [NSMutableDictionary new];
     _groupProfileWhitelistCache = [NSMutableDictionary new];
-    _otherUsersProfileKeyCache = [NSMutableDictionary new];
     _otherUsersProfileAvatarImageCache = [NSCache new];
 
     OWSSingletonAssert();
 
-    // Register this manager with the message sender.
-    // This is a circular dependency.
-    [messageSender setProfilesManager:self];
-
-    // Try to load.
-    _localProfileKey = [self.dbConnection objectForKey:kOWSProfilesManager_LocalProfileSecretKey
-                                            inCollection:kOWSProfilesManager_Collection];
-    if (!_localProfileKey) {
-        // Generate
-        _localProfileKey = [OWSProfilesManager generateLocalProfileKey];
-        // Persist
-        [self.dbConnection setObject:_localProfileKey
-                                forKey:kOWSProfilesManager_LocalProfileSecretKey
-                          inCollection:kOWSProfilesManager_Collection];
+    self.localUserProfile = [self getOrCreateUserProfileForRecipientId:kNSNotificationName_LocalProfileUniqueId];
+    OWSAssert(self.localUserProfile);
+    if (!self.localUserProfile.profileKey) {
```

Comment:
I had it that way originally (for those reasons), but moved to this approach because I wanted to guarantee that this property is a) populated before it is accessed b) always populated on the main thread c) can be accessed synchronously from any thread.